org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -242,9 +242,32 @@ private List<WorkspaceFolder> getRelevantWorkspaceFolders() {
 	 * @throws IOException
 	 */
 	public synchronized void start() throws IOException {
+		start(false);
+	}
+
+	/**
+	 * Restarts a language server. If language server is not started, calling this
+	 * method is the same as calling {@link #start()}.
+	 *
+	 * @throws IOException
```

Comment:
Please add `@since 0.18`, while LSP4E is not best at API versioning, those are getting stable and we should try to leverage good practices whenever prossible.

Reply:
Okay, this makes sense. I wasn't aware that LanguageServerWrapper is API, using it from our cdt plugin gives discouraged access warning.

Reply:
> I wasn't aware that LanguageServerWrapper is API

As mentioned above, LSP4E is not really clean in term of API ;) But lately it seems like there is a consensus that LanguageServers and LanguageServerWrapper can be considered as API.